### ISSUE TITLE
bug: false positives for s3 access block referring to bucket by id

### ISSUE THREAD

**Issue by sergei-ivanov** [aquasecurity/tfsec#1431]
**Describe the bug**

TL;DR
TFsec flags up false positives for the checks:

- `aws-s3-specify-public-access-block`
- `aws-s3-no-public-buckets`
- `aws-s3-ignore-public-acls`
- `aws-s3-block-public-policy`
- `aws-s3-block-public-acls`

... if `aws_s3_bucket_public_access_block` refers to the bucket by `id` attribute rather than by `bucket` attribute.

Longer read:
Terraform provider for AWS returns identical values for `id` and `bucket` attributes on `aws_s3_bucket` resource. As a result, other resources refer to the buckets by either of those attributes. TFsec only seems to resolve references when the `bucket` attribute is being used, e.g. `aws_s3_bucket.example.bucket`. I also figured out that the problem is related to using indexed or keyed resources. For singular resources both `id` and `bucket` references are accepted.
Curiously enough, all "good" examples for TFsec checks show references by id, [for example](https://aquasecurity.github.io/tfsec/v1.0.1/checks/aws/s3/block-public-acls/#secure-example).

Question for discussion:
Since `id` is considered to be an "internal" resource attribute, its use is generally being discouraged by provider authors. Should there be a separate check for reference by `id` that would suggest using equivalent public attribute, e.g. `bucket` for the case of `aws_s3_bucket` resource?

**To Reproduce**

```hcl
variable "enabled" {
  type    = bool
  default = true
}

# tfsec:ignore:aws-s3-enable-bucket-logging
resource "aws_s3_bucket" "this" {
  count = var.enabled ? 1 : 0

  bucket = "test"
  acl    = "private"

  versioning {
    enabled = true
  }

  server_side_encryption_configuration {
    rule {
      apply_server_side_encryption_by_default {
        sse_algorithm = "AES256"
      }
    }
  }
}

resource "aws_s3_bucket_public_access_block" "this" {
  count = var.enabled ? 1 : 0

  bucket                  = aws_s3_bucket.this[0].id  ## when changed to `bucket`, all checks pass
  block_public_acls       = true
  block_public_policy     = true
  ignore_public_acls      = true
  restrict_public_buckets = true
}
```

**Expected behavior**
No problems detected

**Desktop (please complete the following information):**
 
```
$ uname -irs
Linux 5.4.0-94-generic x86_64

$ tfsec --version
1.0.1

$ terraform --version
Terraform v1.1.4
on linux_amd64
```

**Comment by sergei-ivanov** [comment]
Hi @liamg ,
I can confirm that #1431 and #1432 are fixed, but only when I run TFsec in the same directory as the module. Due to #1452 I am still seeing failed checks for `aws-s3-specify-public-access-block` if I run from the top level directory.
